common/utils.py

```diff
@@ -0,0 +1,91 @@
+from six.moves import configparser
+import logging
+import os
+import requests
+
+
+def get_setting(config_file, config_name, setting, default=None):
+    config = configparser.SafeConfigParser()
+    try:
+        config.read(config_file)
+        return config.get(config_name, setting)
+    except Exception:
+        return default
+
+
+def configure_logging(filename): #get_setting('logfile', default_logfile)
+    CONFIG = {
+        'version': 1,
+        'disable_existing_loggers': False,
+        'formatters': {
+            'default': {
+                'format': '%(levelname)-8s  %(asctime)s  %(filename)s:%(lineno)-4s %(message)s',
+                'datefmt': '%Y-%m-%d %H:%M:%S',
+            },
+        },
+        'handlers': {
+            'console': {
+                'class': 'logging.StreamHandler',
+                'formatter': 'default',
+            },
+            'file': {
+                'class': 'logging.handlers.RotatingFileHandler',
+                'formatter': 'default',
+                'filename': filename,
+                'backupCount': 2,
+                'maxBytes': 10 * 1024,
+            },
+        },
+        'loggers': {
+            'storage': {
+                'level': 'INFO',  # One of INFO, DEBUG, WARNING, ERROR, CRITICAL
+                'handlers': ['console', 'file'],
+            },
+        },
+    }
+    logging.config.dictConfig(CONFIG)
+
+
+def open_pid_file(pid_file, logger=None):
+    try:
+        # Open PID file only if it doesn't exist for read/write
+        f = os.fdopen(os.open(pid_file, os.O_CREAT | os.O_EXCL | os.O_RDWR), 'r+')
+    except OSError,e:
+         if logger:
```

Comment:
It makes more sense to assume the logger exists as a module level variable, rather than passed in. Let's define the logger at the top of the file and not pass it as a parameter to the helper functions.